src/AppBundle/Api/EventProvider.php

```diff
@@ -16,11 +18,14 @@ public function __construct(EventRepository $repository, UrlGeneratorInterface $
         $this->urlGenerator = $urlGenerator;
     }
 
-    public function getUpcomingEvents(): array
+    public function getUpcomingEvents(string $type = null): array
     {
-        $data = [];
+        if ($type && !EventCategories::getCategoryName($type)) {
+            throw new EventException(sprintf('Given event category "%s" is invalid.', $type));
+        }
 
-        foreach ($this->repository->findUpcomingEvents() as $event) {
+        $data = [];
```

Comment:
In PHP 7 a micro optim is to do:
```php
return $data ?? [];
```
and remove this memory call which might be needless.